### PR TITLE
Migrate CI On Github Actions

### PR DESCRIPTION
**Pull request by Mandur**
This Pull request aim to migrate our CI from Azure DevOps to Github Actions. New CI has been tests on forked repository [here](https://github.com/Mandur/iotedge-lorawan-starterkit/actions?query=workflow%3A%22LoRa+CI%22). This was made building on the great work of @MaggieSalak.

The new Ci enables:
* More transparent vision on the quality checks we run to our open source community
* We can interrupt checks on private agent (by rebooting the gh action service)

It has the following issues compared to previous version:
* No task to publish test results at the moment (PublishTestResults). Hence no beautified test results windows, we need to go in the logs.
* Slightly slower (< 5 mins) due to lower optimization than previous CI. I think that can be addressed

How does the PR Workflow works?

* Pr are open and the CI will kick in normally.
* Deployment to the physical environment will be protected by manual validation at **each run**. All the LoRa core team have permission to validate a deployment.
* Once a test pass the label is added to the PR, this test won't run if the ci is reran, to enable rerunning the CI easily on some test that were failed due to interferences. If CI author and maintainer wants to rerun all tests, we just need to remove the labels before rerunning the CI
* Only physical tests can be run by using the RunTestsOnly label

How to trigger the custom workflow?

* There will be a button appearing on the actions tab once this is merged to dev
* There people can decide whether to run physical test only or the full ci, the branch to deploy, and which physical test batch to run
* This build will also be subject to environment approval

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+on: # rebuild any PRs and main branch changes
+  pull_request:
+  push:
+    branches:
+      - master
+      - dev
+  workflow_dispatch:
+    inputs:
+      RunTestsOnly:
+        description: 'Run only tests?'
```
Comment (TechPreacher) [inline .github/workflows/ci.yaml:12]: Why a question?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    name: Run IoT Edge Deployment on ARM gateway
+    steps:  
+      - uses: actions/checkout@v2
+        name: Checkout current branch
+
+      - run: |
+          if [ ${{ github.ref	}} = "refs/heads/dev" ]; then
+          echo "dev"  
+          IMAGE_TAG="$DEV_IMAGE_TAG"
+          elif [ ${{ github.ref	}} = "refs/heads/master" ]; then
```
Comment (TechPreacher) [inline .github/workflows/ci.yaml:176]: Is there a reason for the odd spacing?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+          echo "master"
+          IMAGE_TAG="$MASTER_IMAGE_TAG"
+          elif [ ${{ github.event_name }}	== "pull_request" ]; then
```
Comment (TechPreacher) [inline .github/workflows/ci.yaml:179]: Is there a reason for the odd spacing?

File: LoRaEngine/deployment.test.template.json [hunk 4]
```diff
@@ -143,7 +143,7 @@
           "sensordecodermodule": {
             "type": "docker",
             "settings": {
-              "image": "$CONTAINER_REGISTRY_ADDRESS/decodersample:2.0",
+              "image": "$CONTAINER_REGISTRY_ADDRESS/decodersample:2.0-arm32v7",
```
Comment (TechPreacher) [inline LoRaEngine/deployment.test.template.json:146]: Is it OK to hard-code the version 2.0 here?

**Review by TechPreacher — APPROVED** [review]
Looks good. Mostly cosmetic input.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (Mandur) [reply .github/workflows/ci.yaml:179]: I don't know, locally it's not like this

File: LoRaEngine/deployment.test.template.json (hunk 4 above)
Comment (Mandur) [reply LoRaEngine/deployment.test.template.json:146]: I will remove arm32

**Pull request merged**